language-support/ts/daml-types/index.ts

```diff
@@ -123,7 +153,9 @@ export interface Choice<T extends object, C, R, K = unknown> {
   /**
    * Returns the template to which this choice belongs.
    */
-  template: () => Template<T, K>;
+  template: () => T extends Interface<infer I>
+    ? InterfaceCompanion<T, I & string>
+    : Template<T, K>;
```

Comment:
This doesn't have quite the type-level behavior we want; following up in #14949.